### PR TITLE
For review

### PR DESCRIPTION
**Pull request by Abeelie**
Review Everything

### CODE REVIEW

File: helpers/expressError.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+class ExpressError extends Error {
```
Comment (JBaczuk) [inline helpers/expressError.js:1]: I like the custom Errors

File: routes/pages.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+      const albums = await axios.get(url2);
+
+      const url3 = `http://ws.audioscrobbler.com/2.0/?method=artist.gettoptracks&artist=${artists_input.trim()}&limit=10&api_key=${lastFMKey}&format=json`;
+      const topSongs = await axios.get(url3);
+
+      const url4 = `http://ws.audioscrobbler.com/2.0/?method=artist.getsimilar&artist=${artists_input.trim()}&limit=5&api_key=${lastFMKey}&limit=4&format=json`;
+      const similarArtists = await axios.get(url4);
+
+      return res.render("./artists/artists.html", 
+      {img:artists.data.artists[0].strArtistThumb,
```
Comment (JBaczuk) [inline routes/pages.js:97]: what happens if some of these are undefined? Do you know they will always be defined? e.g. `artists.data.artists[0] = undefined`

File: routes/pages.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    } catch (err) {
+      return next(err);
+    }
+  });
+
+
+
+router.get("/events", async function (req, res, next) {
+  try{
+      let results;
```
Comment (JBaczuk) [inline routes/pages.js:192]:
```suggestion
```

(unused)

File: routes/pages.js [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    } catch (err) {
+      return next(err);
+    }
+  });
+
+
+
+router.get("/news", async function (req, res, next) {
+  try{
+      let results;
```
Comment (JBaczuk) [inline routes/pages.js:208]:
```suggestion
```

**Review by JBaczuk — COMMENTED** [review]
This site is much more than most portfolio projects I see. The design is cool, I like the black, and you chose great photos to go with it.

I'm curious why you chose to use a node.js server with nunchucks as opposed to a single page application with an API or something like next.js which does SSR with react.

Great job storing keys in the environment.

you even have unit tests, nice!

Looks mostly good on mobile

Load times are not bad

- [ ] the search functionality is very limited. It looks like you are just passing the string to the last.fm search parameter of the API URLs. It's nice to see an autocomplete.
  - [ ] If there isn't a facebook/twitter URL you probably shouldn't show a link. E.g. search "boo"
- [ ] Some images get squished on smaller screens. e.g.:
image
- [ ] Clicking the documentary thumbnails gives a lot of 404s
- [ ] I'd recommend a linter and formatted like eslint and prettier
- [ ] favicon.ico doesn't appear to be loading correctly

File: helpers/expressError.js (hunk 1 above)
Comment (Abeelie) [reply helpers/expressError.js:1]: Ok

File: routes/pages.js (hunk 2 above)
Comment (Abeelie) [reply routes/pages.js:97]: I will add a preventative measure

File: routes/pages.js (hunk 3 above)
Comment (Abeelie) [reply routes/pages.js:192]: I will remove unused value

File: routes/pages.js (hunk 4 above)
Comment (Abeelie) [reply routes/pages.js:208]: I will remove unused value

**Comment by Abeelie** [comment]
Thank you for the review

**Pull request closed without merging**